sugar.go

```diff
@@ -279,9 +279,7 @@ func (s *SugaredLogger) log(lvl zapcore.Level, template string, fmtArgs []interf
 	}
 
 	msg := getMessage(template, fmtArgs)
-	if ce := s.base.Check(lvl, msg); ce != nil {
-		ce.Write(s.sweetenFields(context)...)
-	}
+	s.base.Log(lvl, msg, s.sweetenFields(context)...)
```

Comment:
I would revert this and the other change in this file.

The Check version has the advantage that we don't spend CPU on `sweetenFields` unless we're actually going to log the message. So if the log level is too low or the message gets sampled, we won't pay to serialize the fields.

Reply:
Ah, that is a good point. Thank you! Reverted.